### DISCUSSION TITLE
Custom macOS dock icon doesn't persist after terminal closure

### DISCUSSION THREAD

**Discussion by dbozbay** [ghostty-org/ghostty#3630] [Feature Requests, Ideas] [answered]
## Description

When configuring a custom dock icon using the macOS-specific settings in `~/.config/ghostty/config`, the custom icon is only displayed while the terminal is open. Upon closing the terminal, the icon reverts to the default Ghostty i

https://github.com/user-attachments/assets/b8992cad-bde5-4256-8523-b9c466873c4c

con in the dock.

## Steps to reproduce

1. Create/edit `~/.config/ghostty/config`
2. Add the following icon-related settings:
  ```
  macos-icon = custom-style
  macos-icon-frame = beige
  macos-icon-screen-color = cba6f7
  macos-icon-ghost-color = 181825
  ```
4. Open Ghostty terminal (custom icon appears)
5. Close Ghostty terminal
6. Observe that the dock icon reverts to default

## Expected behavior

The custom icon configuration should persist in the dock even when the terminal is closed.

## Actual behavior:
The custom icon is only displayed while the terminal is running and reverts to default upon closure.

## Additional information:
- Tried different `macos-icon-frames` values (`chrome`, `aluminum`, `plastic`) - issue persists
- Tried different `macos-icon-screen-color` and `macos-icon-ghost-color` values - issue persists
- Tested after terminal restart and system reboot
- macOS version: Sequoia 15.2
- Ghostty version: 1.0

**Comment by dbozbay** [comment, marked as answer]
Custom config is for runtime only.

**Reply by rom100main** [reply]
I am confused, what's point to custom it?
